src/Groups/group_constructors.jl

```diff
@@ -0,0 +1,323 @@
+   return FPGroup(GAP.Globals.FreeGroup(n))
+end
+
+function free_group(L::String...)
+   J=GAP.julia_to_gap([GAP.julia_to_gap(x) for x in L])
+   return FPGroup(GAP.Globals.FreeGroup(J))
+end
+
+function free_group(n::Int, s::String)
+   return FPGroup(GAP.Globals.FreeGroup(n,GAP.julia_to_gap(s)))
+end
+
+# FIXME: a function `free_abelian_group` with the same signature is
+# already being defined by Hecke
+#function free_abelian_group(n::Int)
+#  return FPGroup(GAP.Globals.FreeAbelianGroup(n))
+#end
+
+# for the definition of group modulo relations, see the quo function in the sub.jl section
+
+function free_group(G::FPGroup)
+   return FPGroup(GAP.Globals.FreeGroupOfFpGroup(G.X))
+end
+
+################################################################################
+#
+# end FpGroups
+#
+################################################################################
+
+"""
+    dihedral_group(n::Int)
+    dihedral_group(::Type{T}, n::Int)
+return the dihedral group of order `n` of type `T`, where `T` is in {``PcGroup``,``PermGroup``,``FPGroup``}. In the first case, the type is set as ``PcGroup``.
+"""
+function dihedral_group(n::Int)
+  @assert iseven(n)
+  return PcGroup(GAP.Globals.DihedralGroup(n))
+end
+
+function dihedral_group(::Type{T}, n::Int) where T <: Group
+  @assert iseven(n)
+  return T(GAP.Globals.DihedralGroup(_gap_filter(T), n))
+end
+
+"""
+    quaternion_group(n::Int)
+    quaternion_group(::Type{T}, n::Int)
+return the quaternion group of order `n` of type `T`, where `T` is in {``PcGroup``,``PermGroup``,``FPGroup``,``MatrixGroup``}. In the first case, the type is set as ``PcGroup``.
+"""
+function quaternion_group(n::Int)
+  @assert iszero(mod(n, 4))
+  return PcGroup(GAP.Globals.QuaternionGroup(n))
+end
+
+function quaternion_group(::Type{T}, n::Int) where T <: Group 
+   @assert iszero(mod(n, 4))
+  return T(GAP.Globals.QuaternionGroup(_gap_filter(T), n))
+end
+
+function isquaternion_group(G::Group)
+  return GAP.Globals.IsQuaternionGroup(G.X)
+end
+
+################################################################################
+#
+# start isometry groups
+#
+################################################################################
+
+function GL(n::Int, q::Int)
+  return MatrixGroup(GAP.Globals.GL(n, q))
+end
+
+function GL(::Type{T}, n::Int, q::Int) where T <: Group
+  return T(GAP.Globals.GL(_gap_filter(T),n, q))
+end
+
+function SL(n::Int, q::Int)
+  return MatrixGroup(GAP.Globals.SL(n, q))
+end
+
+function SL(::Type{T}, n::Int, q::Int) where T <: Group
+  return T(GAP.Globals.SL(_gap_filter(T), n, q))
+end
+
+function symplectic_group(n::Int, q::Int)
+  return MatrixGroup(GAP.Globals.Sp(n, q))
+end
+
+function symplectic_group(::Type{T}, n::Int, q::Int) where T <: Group
+  return T(GAP.Globals.Sp(_gap_filter(T), n, q))
+end
+
+function unitary_group(n::Int, q::Int)
+  return MatrixGroup(GAP.Globals.GU(n, q))
+end
+
+function unitary_group(::Type{T}, n::Int, q::Int) where T <: Group
```

Comment:
Hmm, why isn't this name `GU`? Conversely, why is `SL` not named `special_linear_groups`? I think it should be either the one, or the other, or both (with the short one an alias for the long one), but in any case, done consistently.